broker/src/main/java/org/apache/rocketmq/broker/controller/ReplicasManager.java

```diff
@@ -347,6 +351,17 @@ private void schedulingSyncBrokerMetadata() {
                 }
             } catch (final Exception e) {
                 LOGGER.warn("Error happen when get broker {}'s metadata", this.brokerConfig.getBrokerName(), e);
+                if (e instanceof MQBrokerException) {
+                    int code = ((MQBrokerException) e).getResponseCode();
+                    if (code == CONTROLLER_BROKER_METADATA_NOT_EXIST) {
+                        try {
+                            registerBrokerToController();
+                            TimeUnit.SECONDS.sleep(2);
+                        } catch (InterruptedException ignore) {
+
+                        }
+                    }
+                }
```

Comment:
How about catching  MQBrokerException before Exception？ In addition, could you rebase or merge the latest develop branch to this pull request？ We add new rules for merging

Reply:
yes I will do this